server/cpho/models/indicators.py

```diff
@@ -264,6 +276,38 @@ def __str__(self):
             ]
         )
 
+    def get_relevant_periods(self):
+        # filter Period. to years that have data or have the same type as the indicator's relevant_period_types
+        from .lookups import Period
+
+        globally_relevant = Period.get_currently_relevant_periods()
+
+        # if there isn't a "preference" set, return all relevant periods
+        if not self.relevant_period_types:
+            return globally_relevant
+
+        relevant = []
+        for period in globally_relevant:
+            if (
+                period.quarter
+                and "fiscal_quarters" in self.relevant_period_types
+            ):
+                relevant.append(period)
+            elif (
+                period.year_type == Period.FISCAL_YEAR_TYPE
+                and period.quarter is None
+                and "fiscal_years" in self.relevant_period_types
+            ):
+                relevant.append(period)
+            elif (
+                period.year_type == Period.CALENDAR_YEAR_TYPE
+                and period.quarter is None
+                and "calendar_years" in self.relevant_period_types
+            ):
+                relevant.append(period)
+
+        return relevant
```

Comment:
Note that if the options are left empty, it doesn't do any filtering at all, so this shouldn't really break anything